frontend/src/components/Header.jsx

```diff
@@ -21,6 +21,16 @@ class Header extends React.Component {
     );
   }
 
+  search(event) {
+    const commit = event.target.value.trim();
+    this.props.setSearchQuery && this.props.setSearchQuery(commit);
```

Comment:
Seems that setSearchQuery is always set. Except maybe in tests? In that case, the tests should probably be changed to pass a no-op function instead of the code accepting an undefined setSearchQuery prop. Changing the code like this implies that there are normal cases where setSearchQuery isn't set.

Reply:
The first time that /search/commit/<commitSha> is loaded, setSearchQuery is undefined.  setSearchQuery is only set after the search page is loaded, and the user starts typing search terms.  This check is to make sure that an api call is made during the initial load.